apps/api/src/plugins/auth0.ts

```diff
@@ -0,0 +1,18 @@
+import {FastifyAuth0VerifyOptions} from 'fastify-auth0-verify';
+import fp from 'fastify-plugin';
+
+export default fp<FastifyAuth0VerifyOptions>(async fastify => {
+  await fastify.register(import('fastify-auth0-verify'), {
+    domain: process.env.DOMAIN_AUTH0,
+    secret: process.env.CLIENT_SECRET_AUTH,
+    audience: process.env.AUDIENCE_AUTH0,
+  });
+
+  await fastify.addHook('onRequest', async (request, reply) => {
+    try {
+      await request.jwtVerify();
```

Comment:
Instead of using an hook for each request, here we could try the `authenticate` decorator that the plug-in provides 🤔

It also should be wrapped in a function which will automatically sync the user in the db if not present